스터디 피드백 반영

snwlee opened the pull request:
[♻️ refactor: Comment Domain 객체 피드백 수정](https://github.com/TDD-Study-Though-Young-Man/board-test-homework/commit/56f98d7ede2e109e43e4027f55de2a1eb54fa37d) 

- 스터디 피드백 반영
- Builder public -> private
- 연관관계 매핑 관련 데이터 위치 수정
- Comment Domain 객체 삭제 메서드를 삭제와 복구의 의미에 맞게 2개로 분리

[♻️ refactor: Comment repository 테스트 코드 수정](https://github.com/TDD-Study-Though-Young-Man/board-test-homework/commit/4eb094080f1f67150b88d989bbf7080b66801aa0) 

- 스터디 피드백 반영
- 가독성을 위해 하나의 테스트 메서드 내 테스트 내용을 모두 포함하도록 수정
- `@BeforeEach` 제거

snwlee committed:
⭐ feat: Comment 엔티티 추가

- Comment 엔티티 추가
- Post 임시 엔티티 추가
- Member 엔티티 수정

snwlee committed:
🚦 test: CommentRepository Test 작성

- Comment Persistence Layer Test 작성
- 임시 PostRepository 추가
- Comment, Post 엔티티 PK 필드 명 수정

snwlee committed:
♻️refactor: CommentRepositoryTest given 중복 제거

- 테스트마다 작성했던 동일한 given 내용을 @BeforeEach를 사용한 setup 메서드로 통일

snwlee committed:
⭐ feat: Comment Dto 생성

- CreateCommentRequest, ModifyCommentRequest, CommentResponseDto 생성
- Comment of 메서드 생성
- 임시 PostRepository 생성
- 임시 CommentService 생성

snwlee committed:
♻️ refactor: Comment Domain 객체 피드백 수정

- 스터디 피드백 반영
- Builder public -> private
- 연관관계 매핑 관련 데이터 위치 수정
- Comment Domain 객체 삭제 메서드를 삭제와 복구의 의미에 맞게 2개로 분리

snwlee committed:
♻️ refactor: Comment repository 테스트 코드 수정

- 스터디 피드백 반영
- 가독성을 위해 하나의 테스트 메서드 내 테스트 내용을 모두 포함하도록 수정
- @BeforeEach 제거

snwlee committed:
Merge branch 'TDD-Study-Though-Young-Man:dev-sw.lee' into dev-sw.lee

ch4570 commented on src/test/java/com/member/homework/repository/comment/CommentRepositoryTest.java:
```diff
@@ -0,0 +1,76 @@
+    void 기존_댓글을_수정할_수_있어야_한다() {
+        //given
+        Comment comment = Comment.builder()
+                .author("테스트 댓글 작성자")
```
두 번 이상 사용되는 값의 경우, 변수로 뽑아서 사용해보는 것은 어떨까요?

snwlee replied:
저도 그게 좋다고 생각하여 
초반엔 변수로 처리하였는데
원내대표님께서 가독성이 더 중요하다고 피드백 주셔서
저도 생각이 바뀌었습니다 👍

ch4570 commented on src/main/java/com/member/homework/service/comment/dto/ModifyCommentRequest.java:
```diff
@@ -0,0 +1,4 @@
+package com.member.homework.service.comment.dto;
+
+public record ModifyCommentRequest(Long commentId, String content, boolean deleteYn) {
```
record 클래스를 작성할때 통일성을 위해 괄호를 벌리고, 밑에 필드를 나열하는 표준을 사용하시면 더 좋을 것 같아요!

snwlee replied:
답변 감사드립니다!
해당 부분 수정하여 반영하겠습니다!

[반영 커밋 - 9aa231cbe1a23632b97a6b394e44229df94a6f0f](https://github.com/TDD-Study-Though-Young-Man/board-test-homework/pull/13/commits/9aa231cbe1a23632b97a6b394e44229df94a6f0f)

ch4570 commented:
리뷰 완료하였습니다:)
고생 많으셨어요 😀

featureofme commented:
민재님이 리뷰를 꼼꼼하게 잘 달아주셨습니다.
저는 음 다른거는 다 좋고 @DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class) 요거 한 줄만 클래스 상단부에 추가 부탁드립니다! 고생하셨어요 :)!

snwlee committed:
♻️ refactor: PR 피드백 반영 - CommentRepositoryTest, ModifyCommentRequest 수정

- CommentRepositoryTest @DisplayNameGeneration(DisplayNameGenerator.ReplaceUnderscores.class) 추가
- record 클래스 통일성을 위해 ModifyCommentRequest 필드 위치 수정
